src/libtls/tests/suites/test_socket.c

```diff
@@ -407,6 +407,21 @@ static void run_echo_client(echo_server_config_t *config)
 	server->destroy(server);
 }
 
+/**
+ * Configuration factory
+ */
+static void config_factory(tls_version_t version, uint16_t port, bool cauth,
+						   echo_server_config_t **config)
```

Comment:
We generally use the regular constructor pattern for such functions (the verb probably more often at the beginning because it's a static helper), i.e. something like
```suggestion
static echo_server_config_t *create_config(tls_version_t version,
										   uint16_t port, bool cauth)
```